docs/_posts/murat-gunay/2021-12-26-xlm_roberta_large_token_classifier_hrl_xx.md

```diff
@@ -0,0 +1,125 @@
+{% include programmingLanguageSelectScalaPythonNLU.html %}
+```python
+documentAssembler = DocumentAssembler()\
+      .setInputCol("text")\
+      .setOutputCol("document")
+
+sentenceDetector = SentenceDetectorDLModel.pretrained("sentence_detector_dl", "xx")\
+       .setInputCols(["document"])\
+       .setOutputCol("sentence")
+
+tokenizer = Tokenizer()\
+      .setInputCols(["sentence"])\
+      .setOutputCol("token")
+
+tokenClassifier = XlmRoBertaForTokenClassification.pretrained("xlm_roberta_large_token_classifier_hrl", "xx"))\
+  .setInputCols(["sentence",'token'])\
+  .setOutputCol("ner")
+
+ner_converter = NerConverter()\
+      .setInputCols(["sentence", "token", "ner"])\
+      .setOutputCol("ner_chunk")
+      
+nlpPipeline = Pipeline(stages=[documentAssembler, sentenceDetector, tokenizer, tokenClassifier, ner_converter])
+
+empty_data = spark.createDataFrame([[""]]).toDF("text")
+
+model = nlpPipeline.fit(empty_data)
+text = """يمكنكم مشاهدة أمير منطقة الرياض الأمير فيصل بن بندر بن عبد العزيز في كل مناسبة وافتتاح تتعلق بمشاريع التعليم والصحة وخدمة الطرق والمشاريع الثقافية في منطقة الرياض."""
+result = model.transform(spark.createDataFrame([[text]]).toDF("text"))
+```
+```scala
+val documentAssembler = DocumentAssembler()
+      .setInputCol("text")
+      .setOutputCol("document")
+
+val sentenceDetector = SentenceDetectorDLModel.pretrained("sentence_detector_dl", "xx")
+       .setInputCols(Array("document"))
+       .setOutputCol("sentence")
+
+val tokenizer = Tokenizer()
+      .setInputCols(Array("sentence"))
+      .setOutputCol("token")
+
+val tokenClassifier = XlmRoBertaForTokenClassification.pretrained("xlm_roberta_large_token_classifier_hrl", "xx"))\
+  .setInputCols(Array("sentence","token"))\
+  .setOutputCol("ner")
+
+ner_converter = NerConverter()\
+      .setInputCols(Array("sentence", "token", "ner"))\
+      .setOutputCol("ner_chunk")
+      
+val pipeline = new Pipeline().setStages(Array(documentAssembler, sentenceDetector, tokenizer, tokenClassifier, ner_converter))
+
+val example = Seq.empty["يمكنكم مشاهدة أمير منطقة الرياض الأمير فيصل بن بندر بن عبد العزيز في كل مناسبة وافتتاح تتعلق بمشاريع التعليم والصحة وخدمة الطرق والمشاريع الثقافية في منطقة الرياض."].toDS.toDF("text")
+
+val result = pipeline.fit(example).transform(example)
+```
+</div>
+
+## Results
+
+```bash
++--------------------------+---------+
+|chunk                     |ner_label|
++--------------------------+---------+
+|الرياض                    |PER      |
+|فيصل بن بندر بن عبد العزيز|LOC      |
+|الرياض                    |LOC      |
++--------------------------+---------+
+```
+
+{:.model-param}
+## Model Information
+
+{:.table-model}
+|---|---|
+|Model Name:|xlm_roberta_large_token_classifier_hrl|
+|Compatibility:|Spark NLP 3.3.4+|
+|License:|Open Source|
+|Edition:|Official|
+|Input Labels:|[sentence, token]|
+|Output Labels:|[ner]|
+|Language:|xx|
+|Size:|1.8 GB|
+|Case sensitive:|true|
+|Max sentense length:|256|
+
+## Data Source
+
```

Comment:
Can you include this as well? I think it's super relevant information.

Language | Dataset
-- | --
Arabic | ANERcorp
German | conll 2003
English | conll 2003
Spanish | conll 2002
French | Europeana Newspapers
Italian | Italian I-CAB
Latvian | Latvian NER
Dutch | conll 2002
Portuguese | Paramopama + Second Harem
Chinese | MSRA

Reply:
@josejuanmartinez 
we took this model from hugging face, you know. I think is not ethical to mentioned these data source as if we trained the model. We already show the link on the page to these source.